src/screens/Exchange/ExchangeStatus.js

```diff
@@ -18,50 +18,112 @@
     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
 import * as React from 'react';
+import { Animated } from 'react-native';
 import styled from 'styled-components/native';
-import Animation from 'components/Animation';
 import { fontStyles } from 'utils/variables';
-import { BoldText } from 'components/Typography';
+import { BaseText } from 'components/Typography';
 import { themedColors } from 'utils/themes';
 
 const Status = styled.View`
   flex-direction: row;
-  height: 50px;
-  justify-content: flex-start;
+  height: 22px;
+  justify-content: center;
   align-items: center;
 `;
 
+const StatusText = styled(BaseText)`
+  ${fontStyles.regular};
+  color: ${themedColors.positive};
+  letter-spacing: 0.15px;
+`;
+
 const StatusIcon = styled.View`
   height: 8px;
   width: 8px;
   border-radius: 4px;
   background-color: ${themedColors.positive};
-  position: absolute;
-  top: 7px;
-  left: 7px;
+  margin-left: 8px;
 `;
 
-const StatusText = styled(BoldText)`
-  ${fontStyles.tiny};
-  color: ${themedColors.positive};
-  letter-spacing: 0.15px;
-  margin-top: 2px;
-`;
+const AnimatedStatusIcon = Animated.createAnimatedComponent(StatusIcon);
+const AnimatedStatus = Animated.createAnimatedComponent(Status);
 
-const IconHolder = styled.View`
-  position: relative;
-`;
+type State = {
+  indicatorFadeValue: Animated.Value,
+  statusFadeValue: Animated.Value,
+}
+
+type Props = {
+  isVisible: boolean,
+}
+
+class ExchangeStatus extends React.Component<Props, State> {
+  blinkInterval: IntervalID;
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      indicatorFadeValue: new Animated.Value(1),
+      statusFadeValue: new Animated.Value(props.isVisible ? 1 : 0),
+    };
+  }
+
+  componentDidMount = () => {
+    const { isVisible } = this.props;
+    if (isVisible) this.startBlinking();
+  };
+
+  startBlinking = () => {
+    const { indicatorFadeValue } = this.state;
+    requestAnimationFrame(() => {
```

Comment:
I'm not sure I understand why we want to call `requestAnimationFrame` here; The `requestAnimationFrame` will tell the browser we want to perform an animation, but we're not, we're gonna just setup a timer that's gonna be called on a later tick. Correct?

Reply:
Oh, yeah, you are right. 
Since we have a lot going on on exchange screen, I want to make sure that this blinking animation won't be glitchy. But this request should be inside a loop as well to be effective. I'll update that part. Thanks :)